File: Frontend/interface/hotel.ts
import { User } from "./interface";
import { Province } from "./province";

export interface Hotel {
  id: number;
  name: string;
  address: string;
  user: User; 
  province: Province; 
  createdAt: string; 
  updatedAt: string; 
}

export interface HotelResponse extends Omit<Hotel, "user" | "createdAt" | "updatedAt" | "province">{
  provinceName: string;
  imageURL: string
}

export interface HotelImage {
  id: number;       
  imageUrl: string; 
  hotel: Hotel;     
}

File: Backend/src/main/java/com/example/project_reactnative/model/entity/Hotel.java
package com.example.project_reactnative.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Hotel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String hotelName;

    private String hotelAddress;

    @ManyToOne
    @JoinColumn(name = "manager_id", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "province_id", referencedColumnName = "id")
    private Province province;

    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
